### PR TITLE
[DO NOT MERGE] Only add a reaction if we haven't yet

### LINKED ISSUE #492: Tap reaction to +1 and the use menu for same emoji lets you react twice
API figures this out and negates the second, but our UI +1s twice which is wrong.

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.11.0 (1943)
Device: iPhone 7 Plus (iOS 11.0.1)
TestFlight: false
</pre>
</details>

### PR DESCRIPTION
**Pull request by BasThomas**
Fixes #492 

This now mimics the GitHub behavior: reacting twice removes the first, reacting once... adds it, obviously.

### CODE REVIEW

File: Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift [hunk 1]
```diff
@@ -127,27 +127,35 @@ UICollectionViewDelegateFlowLayout {
...
-        delegate?.didAdd(cell: self, reaction: .confused)
+        addOrRemove(reaction: .confused)
     }
 
     @objc private func onHeart() {
-        delegate?.didAdd(cell: self, reaction: .heart)
+        addOrRemove(reaction: .heart)
+    }
+
+    func addOrRemove(reaction: ReactionContent) {
```
Comment (Sherlouk) [inline Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift]: also known as `toggle` 😅

Comment (BasThomas) [reply Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift]: What's this language you are speaking? 😁

File: Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift [hunk 2]
```diff
@@ -127,27 +127,35 @@ UICollectionViewDelegateFlowLayout {
...
+        toggle(reaction: .confused)
     }
 
     @objc private func onHeart() {
-        delegate?.didAdd(cell: self, reaction: .heart)
+        toggle(reaction: .heart)
+    }
+
+    func toggle(reaction: ReactionContent) {
+        if let didReact = reactions.first(where: { $0.content == reaction })?.viewerDidReact, didReact == true {
```
Comment (rnystrom) [inline Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift]: I'm not crazy about this. We're putting biz logic into the view here.

Also, I think the menu should be an additive action only, and no-op if the reaction is already there (per #492).

I think just a `guard` checking that the reaction doesn't exist yet would be good here:

https://github.com/rnystrom/GitHawk/blob/master/Classes/Issues/Comments/IssueCommentSectionController.swift#L256

Comment (BasThomas) [reply Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift]: Agree on the biz logic. The no-op would be a deviation from how GitHub does it now, though. That ok?

Comment (rnystrom) [reply Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift]: Oooh didn't realize that's what GH does. Hmm. Maybe let's stick w/ how GH does it, just move the biz logic into the controller?

**Comment by BasThomas** [comment]
This now totally no-ops, but wanted to add the addition of the delegate. Talked with Ryan and he'll take a look at this soon 👍

**Pull request closed without merging**